src/server/compaction_operation.h

```diff
@@ -70,6 +70,33 @@ class delete_key : public compaction_operation
 
 private:
     FRIEND_TEST(delete_key_test, filter);
+    FRIEND_TEST(compaction_filter_operation_test, all_rules_match);
+};
+
+enum update_ttl_op_type
+{
+    UTOT_FROM_NOW,
+    UTOT_FROM_CURRENT,
+    UTOT_TIMESTAMP,
+    UTOT_INVALID,
+};
```

Comment:
I recommend adding some comments for each type.

Reply:
done